src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -445,14 +445,15 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
     def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
         assert self.TYPE == daemon_spec.daemon_type
         deps: List[str] = []
-        hostnames: List[str] = []
-        for dd in self.mgr.cache.get_daemons_by_service('mgr'):
-            assert dd.hostname is not None
-            addr = self.mgr.inventory.get_addr(dd.hostname)
-            hostnames.append(addr)
+        daemons = self.mgr.cache.get_daemons_by_service('loki')
+        loki_host = ''
+        assert daemons is not None
+        if daemons != []:
```

Comment:
```suggestion
        if not daemons:
```
better way to check for empty list

Reply:
```suggestion
        if daemons:
```
for empty list check, it should be like this, isn't it?

Reply:
ah, yeah. My mistake. `if daemons` is correct for checking for nonempty list

Reply:
appreciate for your review